packages/api/src/mails/MailchimpMailProvider.ts

```diff
@@ -0,0 +1,113 @@
+import {Mandrill} from 'mandrill-api'
+import crypto from 'crypto'
+
+import {
+  BaseMailProvider,
+  MailLogStatus,
+  MailProviderProps,
+  SendMailProps,
+  WebhookForSendMailProps
+} from './mailProvider'
+import {MailLogState} from '../db/mailLog'
+
+export interface MailchimpMailProviderProps extends MailProviderProps {
+  readonly apiKey: string
+  readonly baseURL: string
+  readonly webhookEndpointSecret: string
+  readonly fromAddress: string
+}
+
+interface VerifyWebhookSignatureProps {
+  signature: string
+  url: string
+  params: Record<string, any>
+}
+
+function mapMandrillEventToMailLogState(event: string): MailLogState | null {
+  switch (event) {
+    case 'send':
+      return MailLogState.Delivered
+    case 'deferral':
+      return MailLogState.Deferred
+    case 'hard_bounce':
```

Comment:
what happens in the case of `hard_bounce` ?

Reply:
https://stackoverflow.com/questions/13207927/switch-statement-multiple-cases-in-javascript